### PR TITLE
Feature/sc 131255 osisoft add timers

### PR DESCRIPTION
**Pull request by alexbourret**

### CODE REVIEW

File: python-connectors/pi-system_event-frames-search/connector.py [hunk 1]
```diff
@@ -105,13 +113,17 @@ def generate_rows(self, dataset_schema=None, dataset_partitioning=None,
                                     if isinstance(batch_row.get("Value"), dict):
                                         value = batch_row.pop("Value", {})
                                         batch_row.update(value)
+                                    self.yields_timer.start()
                                     yield batch_row
+                                    self.yields_timer.stop()
```
[the remark this answers is not in the document]
Comment (alexbourret) [reply python-connectors/pi-system_event-frames-search/connector.py:118]: The use timers will probably be removed from the plugin once everyone is happy with the performances, so I'd rather not add a wrapping around rows generation just for this feature...

File: python-lib/osisoft_plugin_common.py [hunk 2]
```diff
@@ -345,3 +345,65 @@ def is_reached(self):
             return False
         self.counter += 1
         return self.counter > self.records_limit
+
+
+class PerformanceTimer():
```
[the remark this answers is not in the document]
Comment (alexbourret) [reply python-lib/osisoft_plugin_common.py:350]: done in [7ccc330](https://github.com/dataiku/dss-plugin-osisoft/pull/35/commits/7ccc330a04d38b72e4de072a1bf05b2444e7ed90)

File: python-lib/osisoft_plugin_common.py [hunk 3]
```diff
@@ -345,3 +345,65 @@ def is_reached(self):
...
+    def __init__(self):
+        self.slowest_events = []
+        self.slowest_times = []
+        self.total_duration = 0
+        self.number_events = 0
+        self.current_event = None
```
[the remark this answers is not in the document]
Comment (alexbourret) [reply python-lib/osisoft_plugin_common.py:356]: done in [7ccc330](https://github.com/dataiku/dss-plugin-osisoft/pull/35/commits/7ccc330a04d38b72e4de072a1bf05b2444e7ed90)

File: python-lib/osisoft_plugin_common.py [hunk 4]
```diff
@@ -345,3 +345,65 @@ def is_reached(self):
...
+
+    def start(self, event_id=None):
+        self.start_time = float(time.time())
+        self.number_events += 1
+        self.current_event = event_id
+
+    def stop(self):
+        self.stop_time = float(time.time())
+        duration = self.stop_time - self.start_time
```
[the remark this answers is not in the document]
Comment (alexbourret) [reply python-lib/osisoft_plugin_common.py:365]: done in [7ccc330](https://github.com/dataiku/dss-plugin-osisoft/pull/35/commits/7ccc330a04d38b72e4de072a1bf05b2444e7ed90)

File: python-lib/osisoft_plugin_common.py [hunk 5]
```diff
@@ -345,3 +345,65 @@ def is_reached(self):
...
+        self.total_duration += duration
+        if self.current_event:
+            self.add_to_record(duration)
+
+    def add_to_record(self, duration):
```
[the remark this answers is not in the document]
Comment (alexbourret) [reply python-lib/osisoft_plugin_common.py:370]: done in [7ccc330](https://github.com/dataiku/dss-plugin-osisoft/pull/35/commits/7ccc330a04d38b72e4de072a1bf05b2444e7ed90)

File: python-connectors/pi-system_event-frames-search/connector.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (liamlynch-data) [reply python-connectors/pi-system_event-frames-search/connector.py:118]: @alexbourret sure, if it's not a permanent thing than maybe it's not worth changing that (but it wouldn't make make life more complicated really AFAIK)

**Review by liamlynch-data — APPROVED** [review]

**Comment by liamlynch-data** [comment]
All good - retested on downstream branch where the changes were made 👍

**Pull request merged**